library/Icinga/Web/StyleSheet.php

```diff
@@ -137,17 +144,20 @@ protected function collect()
             }
         }
 
-        if ($theme && $theme !== self::DEFAULT_THEME) {
-            if (($pos = strpos($theme, '/')) !== false) {
-                $moduleName = substr($theme, 0, $pos);
-                $theme = substr($theme, $pos + 1);
-                if ($mm->hasLoaded($moduleName)) {
-                    $module = $mm->getModule($moduleName);
-                    $this->lessCompiler->setTheme($module->getCssDir() . '/themes/' . $theme . '.less');
-                }
-            } else {
-                $this->lessCompiler->setTheme($this->pubPath . '/css/themes/' . $theme . '.less');
+        if ($themePath = self::getThemeFile($theme)) {
+                $this->lessCompiler->setTheme($themePath);
+        }
+
+        if ($user = Auth::getInstance()->getUser()) {
+            $mode = $user->getPreferences()->getValue('icingaweb', 'mode', self::DEFAULT_MODE);
+
+            if ($mode === null) {
```

Comment:
I think I now understand why you defined `none` as default above previously. Because currently `none` still isn't applied, as `null` is only returned if you don't define a default value above.

But I'm not that fond of using just the fact that nothing is configured as reason for the `none` mode anyway. Configuration may not only be written by the `PreferenceForm`, but also by other tools. These tools probably won't verify that a theme supports modes at all. Also, it's possible to define a global default theme (application-wide) so then it's not under control of the user and the theme may also one that doesn't support any modes.

You really need to perform the same check as in the `PreferenceForm` (the regex) for the theme that is being applied above.